src/service/irc/IrcService.cpp

```diff
@@ -111,68 +111,68 @@ bool IrcService::onEvent(std::shared_ptr<IEvent> event) {
+                    IrcChannelListing& channel = server.addChannel(channelName,
+                                                                channelStore.getTopic(),
+                                                                channelStore.getDisabled());
+                    for (auto& userPair : channelStore.getUsers()) {
+                        string username = userPair.second.getNick();
+                        channel.addUser(username, "");
+                    }
                 }
             }
-        }
-
-        cout << "[US] Sending chat listing" << endl;
-        appQueue->sendEvent(listing);
-    } else if (type == EventQuerySettings::uuid) {
-        auto query = event->as<EventQuerySettings>();
-
-        auto listing = make_shared<EventIrcSettingsListing>(userId, query->getData());
 
-        std::list<lock_guard<mutex>> locks;
-        // lock all to assure correct results
-        for (auto& cxnPair : ircConnections) {
-            auto& connection = cxnPair.second;
-            locks.emplace_back(connection.getChannelLoginDataMutex());
+            cout << "[US] Sending chat listing" << endl;
+            appQueue->sendEvent(listing);
         }
+        else if (query->getType() == EventQueryType::Settings) {
+            auto listing = make_shared<EventIrcSettingsListing>(userId, query->getData());
 
-        for (auto& cxnPair : ircConnections) {
-            auto& connection = cxnPair.second;
-            auto& serverConfiguration = listing->addServer(connection.getServerId(),
-                                                           connection.getServerName());
-            auto& connectionServerConfiguration = connection.getServerConfiguration();
-            for (auto& nick : connectionServerConfiguration.getNicks())
-                serverConfiguration.addNick(nick);
-            for (auto& hostConfiguration : connectionServerConfiguration.getHostConfigurations()) {
-                serverConfiguration.addHostConfiguration(hostConfiguration.getHostName(),
-                                                         hostConfiguration.getPort(),
-                                                         hostConfiguration.getPassword(),
-                                                         hostConfiguration.getIpV6(),
-                                                         hostConfiguration.getSsl());
+            std::list<lock_guard<mutex>> locks;
+            // lock all to assure correct results
+            for (auto& cxnPair : ircConnections) {
+                auto& connection = cxnPair.second;
+                locks.emplace_back(connection.getChannelLoginDataMutex());
             }
-        }
 
-        appQueue->sendEvent(listing);
+            for (auto& cxnPair : ircConnections) {
+                auto& connection = cxnPair.second;
+                auto& serverConfiguration = listing->addServer(connection.getServerId(),
```

Comment:
Here `addServer` returns `IrcServerConfiguration`, which is totally different from `IrcServerListing` returned by `addServer` above.